src/common/low_precision_transformations/src/markup_precisions.cpp

```diff
@@ -113,13 +117,21 @@ bool ngraph::pass::low_precision::MarkupPrecisions::run_on_model(const std::shar
                     continue;
                 }
 
-                const pass::low_precision::PrecisionsRestriction::PrecisionsByPorts& precisionsByPorts = it2->second;
-                setRestriction(node, precisionsByPorts);
+                const Restriction::RestrictionByVersion& precisionsByPorts = it2->second;
+                setRestriction(
+                    node,
+                    (precisionsByPorts.precisionsFunction != nullptr) ?
+                        precisionsByPorts.precisionsFunction(node) :
+                        precisionsByPorts.precisions);
```

Comment:
Can we encapsulate this precisions computation logic in `setRestriction` method or in some `Restriction::RestrictionByVersion` method?

Reply:
done, thanks